tests/test_sim_results/test_aggregate_results.py

```diff
@@ -68,17 +74,18 @@ def test_bids_offers_trades_buffer_gets_updated(self):
 
         # check no duplicated entries are added for each timeslot
         for order in ("bids", "offers", "trades"):
-            buffered_orders = set()
+            buff_orders = set()
             for market_stats in market_results_aggr.bids_offers_trades.values():
-                buffered_orders = buffered_orders.union(set(market_stats[order]))
-            assert len(buffered_orders) == 4 * MARKET_TIMESLOTS * ORDERS_PER_MARKET_SLOT
+                buff_orders = buff_orders.union(set(o["uuid"] for o in market_stats[order]))
```

Comment:
Out of curiosity, are there any orders that do not include a uuid ?

Reply:
I'm not sure. I did this to ensure collected orders are unique and the same order is not repeatedly appended as a fault. Why are asking this?

Reply:
OK I see the rationale behind this now. I was just wondering because the old version of the code would end up in exactly the same outcome since the buff_orders is only used for counting the length of the order list. Since the order uuids are created as part of the test using uuid4 there is (almost) never a chance of hash collision, since it ends up to the same outcome though no problem at all keeping it like this.

Reply:
Good. I will add some comments to explain the rationale behind this in the code. Thanks for pointing it out.